Foundation/Decimal.swift

```diff
@@ -340,56 +340,57 @@ extension Decimal : SignedNumeric {
         fatalError()
     }
 
-    public static func +=(_ lhs: inout Decimal, _ rhs: Decimal) {
-        var leftOp = lhs
-        var rightOp = rhs
-        _ = NSDecimalAdd(&lhs, &leftOp, &rightOp, .plain)
+    public static func +=(lhs: inout Decimal, rhs: Decimal) {
+        var rhs = rhs
+        _ = withUnsafeMutablePointer(to: &lhs) {
```

Comment:
Is there a benefit to using `withUnsafeMutablePointer(to:)` here? Seems better to not requite `unsafe` functions if we can avoid using them.

Reply:
`NSDecimalAdd`, etc., are explicitly noted in both the Obj-C and Swift versions to permit the result pointer to be the same as either the left or right operand. Calling the method directly was possible (and indeed what we did in both versions of `Decimal`) until enforcement of the Law of Exclusivity.

The `withUnsafeMutablePointer(to:)` version is safe because, well, that's the documented behavior of the API we call. By using it, we tell the compiler not to deny us based the Law of Exclusivity (a change in this case made by me and reviewed by @phausler, I believe, over in the Darwin overlay). The alternative is an additional copy which may or may not be elided on compilation, which is less optimal for a basic operation.